controllers/ovndbcluster_controller.go

```diff
@@ -458,7 +462,6 @@ func (r *OVNDBClusterReconciler) reconcileNormal(ctx context.Context, instance *
 	if instance.Status.ReadyCount > 0 && len(svcList.Items) > 0 {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
```

Comment:
you are right, this is currently not possible. we changed a while back lib-common on purpose to merge Annotations and Labels with [1], where we hit an issue not merging it because metallb operator started to add an annotation on success and not merging the annotations resulted in renconcile loop.  but you are right, this would prevent updating the dnsDataLabelSelectorValue, if needed. right now I can only think of one use case for a need to change this, when there is the need to run multiple dns servers in a namespace. we probably want to enhance lib-common Template [1] to allow controlling labels/annotations to be merged or not and then use this for dnsdata cm create.

[1] https://github.com/openstack-k8s-operators/lib-common/commit/18d0ea2c06a85f0d9a329c8f7ffbff41afeb0f1c
[2] https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/util/template_util.go#L55

Reply:
Thanks @stuggi for the details. It should be fine for now, If there is a need can be changed later.